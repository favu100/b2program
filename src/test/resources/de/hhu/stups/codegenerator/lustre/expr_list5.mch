MACHINE expr_list5

SEES LibraryLustre

VARIABLES expr_list5_is_initialisation, expr_list5_var_b, expr_list5_var_a, expr_list5_var_x

SETS expr_list5_clocks = {expr_list5_clock___basic__}

INVARIANT ((expr_list5_is_initialisation : expr_list5_clocks --> BOOL) & ((expr_list5_var_b : REF +-> INTEGER) & ((expr_list5_var_a : REF +-> INTEGER) & ((expr_list5_var_x : REF +-> INTEGER) & (((expr_list5_is_initialisation(expr_list5_clock___basic__) = FALSE) => (expr_list5_var_a /= {})) & ((expr_list5_is_initialisation(expr_list5_clock___basic__) = FALSE) => (expr_list5_var_b /= {})))))))

INITIALISATION 
    expr_list5_is_initialisation := (expr_list5_clocks * {TRUE}); 
    expr_list5_var_b := {}; 
    expr_list5_var_a := {}; 
    expr_list5_var_x := {}


OPERATIONS 

    expr_list5_clock_step(expr_list5_param_x) = 
        PRE
            (expr_list5_param_x : REF --> INTEGER)
        THEN
            expr_list5_var_x := expr_list5_param_x; 
            expr_list5_var_b := l_fby_integer(expr_list5_is_initialisation(expr_list5_clock___basic__) |-> {ref |-> 1} |-> expr_list5_var_x); 
            expr_list5_var_a := expr_list5_var_b; 
            expr_list5_is_initialisation(expr_list5_clock___basic__) := FALSE

        END;

        out <-- getA = BEGIN out := expr_list5_var_a END;
        out <-- getB = BEGIN out := expr_list5_var_b END

END