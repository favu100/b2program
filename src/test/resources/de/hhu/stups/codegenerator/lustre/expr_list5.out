{(ref |-> 1)}
{(ref |-> 1)}
{(ref |-> 4)}
{(ref |-> 4)}
